focus-group.js

```diff
@@ -109,3 +107,36 @@ export function menuKeyUX(options) {
     }
   }
 }
+
+export function findGroupNodeByEventTarget(eventTarget) {
+  let supportedRoles = {
+    'menuitem': ["menu", "menubar"],
+    'option': ["listbox"],
+    'tab': ["tablist"]
+  }
+  let itemRole = eventTarget.role
+  let groupRoles = supportedRoles[itemRole]
+  if (!groupRoles) return null
+
+  let result
+  groupRoles.forEach((role) => {
+    let node = eventTarget.closest(`[role=${role}]`)
+    if (node) {
+      result = node
+    }
+  })
+  return result
+}
+
+export function isHorizontalOrientation(group) {
+  let ariaOrientation = group.getAttribute('aria-orientation')
+  if (ariaOrientation === "vertical") return false
+  if (ariaOrientation === "horizontal") return true
+
+  if (!ariaOrientation) {
+    let role = group.role
+    return role === "menubar" || role === "tablist"
+  }
+
+  return false
+}
```

Comment:
Newline at the end of the file is missed.

I can recommend installing EditorConfig extension to your text editor.